pkg/log/multislogger/multislogger.go

```diff
@@ -0,0 +1,86 @@
+package multislogger
+
+import (
+	"context"
+	"log/slog"
+
+	slogmulti "github.com/samber/slog-multi"
+)
+
+type MultiSlogger struct {
+	*slog.Logger
+	handlers map[string]MultiSloggerHandler
+}
+
+type MultiSloggerHandler struct {
+	handler  slog.Handler
+	matchers []func(ctx context.Context, r slog.Record) bool
+}
```

Comment:
It does not need to be exported, it's just a bag to hold handlers and their matchers. IMO, it's cleaner to have a collection of structs instead of the dual maps.

We need to store the matchers because each time a handler is added, we have to rebuild the router. I'm using a the router I found [here](https://github.com/samber/slog-multi/blob/main/router.go), but we could role our own.

Reply:
Right, but why do that? what circumstances lead to wanting to change a handler?